src/early_multicellular_stage/editor/CellBodyPlanEditorComponent.cs

```diff
@@ -390,6 +390,13 @@ public override void OnFinishEditing()
         editedSpecies.UpdateNameIfValid(newName);
 
         behaviourEditor.OnFinishEditing();
+
+        foreach (var cellType in cellTypeSelectionButtons.Values)
```

Comment:
I think this should instead loop through the cell types inside the species to not rely on GUI state being correct (i.e. if a button is missing for some reason or not detected it would still work, and I think going from simulation data -> GUI is a very good rule to follow and not do GUI -> simulation data transfer of information unless really necessary).